### PR TITLE
Migrate Pulsar from Bintray to Maven Central

### PR DESCRIPTION
**Pull request by timmyyuan**
Motivation

Jcenter will close the upload business of Bintray on May 1st and we need to migrate Pulsar to Maven Central. These changes add the required plugins and distributionManagement to achieve this goal.

The main references of this work are:
https://central.sonatype.org/pages/ossrh-guide.html
https://blog.10pines.com/2018/06/25/publish-artifacts-on-maven-central/

### CODE REVIEW

**Comment by vijaykramesh** [comment]
any update here?  bintray had a brown out today to prepare for the 1st...

File: pom.xml [hunk 1]
```diff
@@ -23,13 +23,8 @@
   xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
   <modelVersion>4.0.0</modelVersion>
   <packaging>pom</packaging>
-  <parent>
-    <groupId>org.apache</groupId>
-    <artifactId>apache</artifactId>
-    <version>23</version>
-  </parent>
```
Comment (codelipenghui) [inline pom.xml:30]: Do we need to remove the parent?

**Comment by codelipenghui** [comment]
@vijaykramesh We could complete the migration of maven central in these two days

File: pom.xml (hunk 1 above)
Comment (timmyyuan) [reply pom.xml]: The changes don’t seem to be needed, I have added the parent again.

**Review by timmyyuan — COMMENTED** [review]
@codelipenghui I have checked it again, and now there is no fatal error or unrelated change with the compilation. Only two errors that do not affect compilation still exist:

1. `Error fetching link: ... buildtools/target/apidocs/package-list. Ignored it.`
Seems we generate a empty buiildtools/target/apidocs/package-list

2. `MavenInvocationException: Error when invoking Maven, consult the invoker log file: ...txt`
The error message is
```
buildtools/src/main/java/org/apache/pulsar/tests/xx.java: package org.apache.commons.lang3 does not exist
buildtools/src/main/java/org/apache/pulsar/tests/MockitoCleanupListener.java error: bad use of '>'
```
Seems to be a issue about dependency broken or doclint failure. I tried to investigate both of them but all ended in failure. 

It seems that all the issues lie in the buildtool, do you have any ideas?

File: pulsar-client-all/pom.xml [hunk 2]
```diff
@@ -103,7 +103,7 @@
               <target>
                 <!-- shade the AsyncHttpClient ahc-default.properties files -->
                 <replace token= "org.asynchttpclient."
-                         value="org.apache.pulsar.shade.org.asynchttpclient."
+                         value="io.streamnative.shade.org.asynchttpclient."
```
[the remark this answers is not in the document]
Comment (codelipenghui) [reply pulsar-client-all/pom.xml:106]: We don't need to change the package name for the maven central migration. We are only changing the group ID.

[the remark this answers is not in the document]
Comment (timmyyuan) [reply pulsar-client-all/pom.xml]: Thanks, I revised them back.

**Comment by timmyyuan** [comment]
Done

**Pull request closed without merging**